### PR TITLE
Localization support

### PR DESCRIPTION
**Pull request by andrunko**
This adds localization support to the panel extension and imports translations from eos 3.8.

This branch is on top of https://github.com/endlessm/eos-panel-extension/pull/7 because it would otherwise have a small conflict on `utils.js` and also it already includes the translations for the new strings added there for the hotcorner menu.

https://phabricator.endlessm.com/T30445

### CODE REVIEW

File: .tx/config [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[main]
+host = https://www.transifex.com
+
+[eos-panel-extension.all]
+source_file = locale/eos-panel-extension.pot
```
Comment (wjt) [inline .tx/config]: Based on the other patches which put the translations in 'po/' this cannot be right, surely:

```suggestion
source_file = po/eos-panel-extension.pot
```

File: .tx/config [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+source_lang = en
+file_filter = locale/<lang>.po
```
Comment (wjt) [inline .tx/config]:
```suggestion
file_filter = po/<lang>.po
```

Comment (wjt) [reply .tx/config]: I think you copied these from eos-hack-extension which uses the (unconventional) `locale/` directory.

File: utils.js [hunk 3]
```diff
@@ -22,6 +23,65 @@ const { Gio } = imports.gi;
...
+const Gettext = imports.gettext.domain('eos-panel-extension');
+
+var gettext = Gettext.gettext;
+
+function getSettings() {
+    const dir = PanelExtension.dir.get_child('schemas').get_path();
+    const source = Gio.SettingsSchemaSource.new_from_directory(dir,
+        Gio.SettingsSchemaSource.get_default(), false);
+    if (!source)
+        throw new Error('Error Initializing the thingy.');
```
Comment (wjt) [inline utils.js]: The thingy?

File: utils.js [hunk 4]
```diff
@@ -22,6 +23,65 @@ const { Gio } = imports.gi;
...
+    return new Gio.Settings({ settings_schema: schema });
+}
+
+function getMigrationSettings() {
+    const dir = PanelExtension.dir.get_child('schemas').get_path();
+    const source = Gio.SettingsSchemaSource.new_from_directory(dir,
+        Gio.SettingsSchemaSource.get_default(), false);
+
+    if (!source)
+        throw new Error('Error Initializing the thingy.');
```
Comment (wjt) [inline utils.js]: The thingy!

**Review by wjt — CHANGES_REQUESTED** [review]

File: .tx/config (hunk 1 above)
Comment (andrunko) [reply .tx/config]: Fixed, bad copy paste.

File: .tx/config (hunk 2 above)
Comment (andrunko) [reply .tx/config]: Same.

File: utils.js (hunk 3 above)
Comment (andrunko) [reply utils.js]: https://github.com/endlessm/eos-hack-extension/blob/master/utils.js#L16 - need to check better when copying stuff hehe. Fixed.

File: utils.js (hunk 4 above)
Comment (andrunko) [reply utils.js]: Same.

File: utils.js (hunk 3 above)
Comment (andrunko) [reply utils.js]: Note that this change is  actually from https://github.com/endlessm/eos-panel-extension/pull/7 though (which this PR is on top).

**Comment by andrunko** [comment]
Force pushed to rebase on top of latest changes from #7 which this PR is on top.

**Comment by GeorgesStavracas** [comment]
This PR needs some handholding, but otherwise it LGTM.

**Comment by andrunko** [comment]
Force pushed after rebasing on latest master, no actual change.

**Pull request merged**